src/coreclr/vm/stringliteralmap.cpp

```diff
@@ -577,16 +585,19 @@ void GlobalStringLiteralMap::RemoveStringLiteralEntry(StringLiteralEntry *pEntry
         }
 #endif
 
-        // Release the object handle that the entry was using.
-        STRINGREF *pObjRef = pEntry->GetStringObject();
-        m_PinnedHeapHandleTable.ReleaseHandles((OBJECTREF*)pObjRef, 1);
+        if (pEntry->IsStringFrozen())
```

Comment:
```suggestion
        if (!pEntry->IsStringFrozen())
```
This needs to be other way around

Reply:
Oops, thanks